iroh/src/magicsock.rs

```diff
@@ -184,12 +184,9 @@ pub(crate) struct MagicSock {
     /// Relay datagrams received by relays are put into this queue and consumed by
     /// [`AsyncUdpSocket`].  This queue takes care of the wakers needed by
     /// [`AsyncUdpSocket::poll_recv`].
-    relay_datagrams_queue: Arc<RelayDatagramsQueue>,
-    /// Waker to wake the [`AsyncUdpSocket`] when more data can be sent to the relay server.
-    ///
-    /// This waker is used by [`IoPoller`] and the [`RelayActor`] to signal when more
-    /// datagrams can be sent to the relays.
-    relay_send_waker: Arc<std::sync::Mutex<Option<Waker>>>,
+    relay_datagram_recv_queue: Arc<RelayDatagramRecvQueue>,
```

Comment:
seems inconsistent, one neds an `Arc` and one doesn't?

Reply:
Well, one is a `Sender` which is `Clone`.  The other is a queue.  I don't think this is a big deal, just an artifact of how the channels/queues are built internally.